wcfsetup/install/files/lib/system/database/table/index/DatabaseTableForeignKey.class.php

```diff
@@ -0,0 +1,271 @@
+<?php
+namespace wcf\system\database\table\index;
+use wcf\system\application\ApplicationHandler;
+use wcf\system\database\table\TDroppableDatabaseComponent;
+
+/**
+ * Represents a foreign key of a database table.
+ * 
+ * @author	Matthias Schmidt
+ * @copyright	2001-2019 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core\System\Database\Table\Index
+ * @since	5.2
+ */
+class DatabaseTableForeignKey {
+	use TDroppableDatabaseComponent;
+	
+	/**
+	 * columns affected by the foreign key 
+	 * @var	string[]
+	 */
+	protected $columns;
+	
+	/**
+	 * name of the foreign key
+	 * @var	string
+	 */
+	protected $name;
+	
+	/**
+	 * action executed in referenced table if row is deleted
+	 * @var	null|string
+	 */
+	protected $onDelete;
+	
+	/**
+	 * action executed in referenced table if row is updated
+	 * @var	null|string
+	 */
+	protected $onUpdate;
+	
+	/**
+	 * relevant columns in referenced table
+	 * @var	string[]
+	 */
+	protected $referencedColumns;
+	
+	/**
+	 * name of referenced table
+	 * @var	string
+	 */
+	protected $referencedTable;
+	
+	/**
+	 * valid on delete/update actions
+	 * @var	string[]
+	 */
+	const VALID_ACTIONS = [
```

Comment:
Missing `RESTRICT` and `SET DEFAULT`

Reply:
- `RESTRICT` is a MySQL specific alias of `NO ACTION`.
> NO ACTION: A keyword from standard SQL. In MySQL, equivalent to RESTRICT.
- `SET DEFAULT` is supported by the grammar, but not actually supported:
> SET DEFAULT: This action is recognized by the MySQL parser, but both InnoDB and NDB reject table definitions containing ON DELETE SET DEFAULT or ON UPDATE SET DEFAULT clauses.

Reply:
Additionally, these actions are the same as the ones supported the [SQL parser](https://github.com/WoltLab/WCF/blob/ba2c8f93b2dcb0df8c3281654fb3bbb2102a0163/wcfsetup/install/files/lib/system/database/util/SQLParser.class.php#L122).